astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -1934,6 +1935,49 @@ def test_unique(self, kwargs):
     def test_unique_more_complex(self, kwargs):
         self.check1(np.unique, **kwargs)
 
+    if not NUMPY_LT_2_0:
+
+        @needs_array_function
+        def test_unique_all(self):
+            values, indices, inverse_indices, counts = np.unique(
+                self.q,
+                return_index=True,
+                return_inverse=True,
+                return_counts=True,
+                equal_nan=False,
+            )
+            res = np.unique_all(self.q)
+
+            assert_array_equal(values, res.values)
+            assert_array_equal(indices, res.indices)
+            assert_array_equal(inverse_indices, res.inverse_indices)
+            assert_array_equal(counts, res.counts)
+
+        @needs_array_function
+        def test_unique_counts(self):
+            values, counts = np.unique(self.q, return_counts=True, equal_nan=False)
+            res = np.unique_all(self.q)
+
+            assert_array_equal(values, res.values)
+            assert_array_equal(counts, res.counts)
+
+        @needs_array_function
+        def test_unique_inverse(self):
+            values, inverse_indices = np.unique(
+                self.q, return_inverse=True, equal_nan=False
+            )
+            res = np.unique_all(self.q)
+
+            assert_array_equal(values, res.values)
+            assert_array_equal(inverse_indices, res.inverse_indices)
+
+        @needs_array_function
+        def test_unique_values(self):
+            values = np.unique(self.q, equal_nan=False)
+            res = np.unique_all(self.q)
+
+            assert_array_equal(values, res.values)
```

Comment:
In this case, I think `res` is just the array.

Reply:
good call. Fixed